include/buffer_control_block.h

```diff
@@ -20,25 +23,43 @@ typedef uint32_t buffer_id_t;
 class BufferControlBlock {
 private:
   buffer_id_t id;
-
   // busy lock
   std::mutex mtx;
 
   // condition variable to determine if the buffer needs to be flushed
-  std::condition_variable needs_flushing;
-
-  // file name of backing store, less directory information
-  std::string f_ext;
+  // std::condition_variable needs_flushing;
 
   // how many items are currently in the buffer
-  uint32_t storage_ptr;
+  File_Pointer storage_ptr;
+
+  // where in the file is our data stored
+  File_Pointer file_offset;
```

Comment:
I'm assuming `storage_ptr - file_offset` is the number of bytes we have in the buffer?

Reply:
Yes. file_offset is the position of this buffer within our backing_store file, storage_ptr points to the first empty slot in our buffer.